library/colour.cpp

```diff
@@ -0,0 +1,63 @@
+#include <algorithm>
+#include "colour.h"
+
+namespace OpenApoc {
+
+Colour Colour::FromHtmlName(const UString& name)
+{
+	UString name_lower = name.toLower();
+	if      (name_lower == "white")   return {255, 255, 255};
+	else if (name_lower == "silver")  return {192, 192, 192};
+	else if (name_lower == "gray")    return {128, 128, 128};
+	else if (name_lower == "black")   return {0,   0,   0};
+	else if (name_lower == "red")     return {255, 0,   0};
+	else if (name_lower == "maroon")  return {128, 0,   0};
+	else if (name_lower == "yellow")  return {255, 255, 0};
+	else if (name_lower == "olive")   return {128, 128, 0};
+	else if (name_lower == "lime")    return {0,   255, 0};
+	else if (name_lower == "green")   return {0,   255, 0};
+	else if (name_lower == "aqua")    return {0,   255, 255};
+	else if (name_lower == "teal")    return {0,   128, 128};
+	else if (name_lower == "blue")    return {0,   0,   255};
+	else if (name_lower == "navy")    return {0,   0,   128};
+	else if (name_lower == "fuchsia") return {255, 0,   255};
+	else if (name_lower == "purple")  return {128, 0,   128};
+	// unknown color name
+	return {0, 0, 0};
```

Comment:
Maybe logWarning()? Might be useful for modders etc. to be explicitly told they're doing something unexpected